lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -98,14 +98,16 @@ async function runA11yChecks() {
 }
 /* c8 ignore stop */
 
+/** @implements {LH.Gatherer.FRGathererInstance<'AppCacheManifest'>} */
 class Accessibility extends FRGatherer {
-  /** @type {LH.Gatherer.GathererMeta} */
+  /** @type {LH.Gatherer.GathererMeta<'AppCacheManifest'>} */
   meta = {
+    dependencies: {AppCacheManifest: require('./dobetterweb/appcache.js').GathererSymbol},
```

Comment:
the gatherer symbol will be used by the config for lookup and validation as a typecheck of sorts

Reply:
> Where would we need to use unique symbols to identify dependencies?

It doesn't *have* to be symbols, but symbols were discussed as the consensus favorite over strings-only during design doc discussion as a convenient safeguard while we venture into this world of a more flexible dynamically generated artifact dependencies. The other nice property of symbols is that it mostly forces you to `require` in the thing you're depending on instead of just getting away with `['AppCacheManifest']`

> why does it need to go under AppCacheManifest? Couldn't this just be an array?

2 main reasons, 1 minor reason:

- Readability. It's much clearer to use `context.dependencies.AppCacheManifest` than it is to use `context.dependencies[0]`
- Type challenges. Dynamically crafting `[AppCacheManifest]` is a nightmare compared to `Pick<Artifacts, DaKeys>`
- Extensibility. Separating the name the gatherer calls its own dependency from the name the thing that provides the dependency calls itself allows us to loosen the tight mapping of today selectively in the future if need be.